src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -0,0 +1,234 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.sensors.CANCoder;
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.units.Distance;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.units.Velocity;
+import edu.wpi.first.units.Voltage;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import static edu.wpi.first.units.Units.MetersPerSecond;
+import static edu.wpi.first.units.Units.Volts;
+
+public class SwerveModule {
+
+
+    private static final double DRIVE_GEAR_RATIO = 6.75;
+    private static final double STEER_GEAR_RATIO = 12.8;
+    private static final double WHEEL_RADIUS = 0.0508;
+    private static final double WHEEL_CIRCUMFERENCE = 2*Math.PI*WHEEL_RADIUS;
+
+    private static final double STEER_P = 0.56; // 0.05 //0.052 0053 0.054
+    private static final double STEER_I = 0.00003; //0
+    private static final double STEER_D = 0.01;
+    private static final double STEER_F = 0.;
+
+    private static final double DRIVE_P = 0.0001;//0.0003;
+    private static final double DRIVE_I = 0;//5e-7;
+    private static final double DRIVE_D = 0;//5e-7;
+    private static final double DRIVE_F = 0.00016;//0.0001;
+
+    private final CANSparkMax drive;
+    private final CANSparkMax steer;
+    private final CANCoder canCoder;
```

Comment:
switch to phoenix v6.

read [here](https://github.com/CrossTheRoadElec/Phoenix6-Examples/tree/main/java/CANcoder) and [here](https://v6.docs.ctr-electronics.com/en/stable/docs/migration/migration-guide/index.html)

remember to update the devices themselves to phoenixv6 via the phoenix tuner.